app/clients/sms/aws_pinpoint.py

```diff
@@ -14,19 +14,22 @@ class AwsPinpointClient(SmsClient):
     def init_app(self, current_app, statsd_client, *args, **kwargs):
         self._client = boto3.client("pinpoint-sms-voice-v2", region_name="ca-central-1")
         super(AwsPinpointClient, self).__init__(*args, **kwargs)
-        # super(SmsClient, self).__init__(*args, **kwargs)
         self.current_app = current_app
         self.name = "pinpoint"
         self.statsd_client = statsd_client
 
     def get_name(self):
         return self.name
 
-    def send_sms(self, to, content, reference, multi=True, sender=None):
-        pool_id = self.current_app.config["AWS_PINPOINT_SC_POOL_ID"]
+    def send_sms(self, to, content, reference, multi=True, sender=None, template_id=None):
         messageType = "TRANSACTIONAL"
         matched = False
 
+        if template_id is not None and str(template_id) in self.current_app.config["AWS_PINPOINT_SC_TEMPLATE_IDS"]:
+            pool_id = self.current_app.config["AWS_PINPOINT_SC_POOL_ID"]
+        else:
+            pool_id = self.current_app.config["AWS_PINPOINT_DEFAULT_POOL_ID"]
```

Comment:
What happens if the pool_id is `None`? Is a random `OriginationIdentity` would be used then?

Reply:
if the  `OriginationIdentity` is `None` in the boto call then the call returns an error
```
Invalid type for parameter OriginationIdentity, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
```
But as you mention below, we catch that case in our code and fall back to SNS.